### ISSUE TITLE
OSError: [Errno 86] Bad CPU type in executable: '/Users/jm/.wdm/drivers/geckodriver/macos/v0.29.1/geckodriver'

### ISSUE THREAD

**Issue by justinTM** [SergeyPirogov/webdriver_manager#175]
```shell
jm ~ % python3 --version
Python 3.9.2
jm ~ % python3 -m venv ./my-venv
jm ~ % cd my-venv && source bin/activate
(my-venv) jm my-venv % python --version
Python 3.9.2
(my-venv) jm my-venv % pip -q install --upgrade pip
(my-venv) jm my-venv % pip install selenium     
Collecting selenium
  Using cached selenium-3.141.0-py2.py3-none-any.whl (904 kB)
Collecting urllib3
  Using cached urllib3-1.26.4-py2.py3-none-any.whl (153 kB)
Installing collected packages: urllib3, selenium
Successfully installed selenium-3.141.0 urllib3-1.26.4

(my-venv) jm my-venv % pip install webdriver-manager 
Requirement already satisfied: webdriver-manager in ./lib/python3.9/site-packages (3.3.0)
Requirement already satisfied: configparser in ./lib/python3.9/site-packages (from webdriver-manager) (5.0.2)
Requirement already satisfied: requests in ./lib/python3.9/site-packages (from webdriver-manager) (2.25.1)
Requirement already satisfied: crayons in ./lib/python3.9/site-packages (from webdriver-manager) (0.4.0)
Requirement already satisfied: colorama in ./lib/python3.9/site-packages (from crayons->webdriver-manager) (0.4.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./lib/python3.9/site-packages (from requests->webdriver-manager) (1.26.4)
Requirement already satisfied: chardet<5,>=3.0.2 in ./lib/python3.9/site-packages (from requests->webdriver-manager) (4.0.0)
Requirement already satisfied: idna<3,>=2.5 in ./lib/python3.9/site-packages (from requests->webdriver-manager) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in ./lib/python3.9/site-packages (from requests->webdriver-manager) (2020.12.5)

(my-venv) jm my-venv % python
Python 3.9.2 (v3.9.2:1a79785e3e, Feb 19 2021, 09:09:00) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from selenium import webdriver
>>> from webdriver_manager.firefox import GeckoDriverManager
>>> 
>>> driver = webdriver.Firefox(executable_path=GeckoDriverManager().install())


[WDM] - ====== WebDriver manager ======
[WDM] - Driver [/Users/jm/.wdm/drivers/geckodriver/macos/v0.29.1/geckodriver] found in cache
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jm/my-venv/lib/python3.9/site-packages/selenium/webdriver/firefox/webdriver.py", line 164, in __init__
    self.service.start()
  File "/Users/jm/my-venv/lib/python3.9/site-packages/selenium/webdriver/common/service.py", line 72, in start
    self.process = subprocess.Popen(cmd, env=self.env,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1823, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 86] Bad CPU type in executable: '/Users/jm/.wdm/drivers/geckodriver/macos/v0.29.1/geckodriver'
>>> 
```

**Comment by justinTM** [comment]
macOS Catalina 10.15.7

**Comment by gbrabelo** [comment]
same problem here. It seems related to mac M1 chips being available with a similar name now.

**Comment by SergeyPirogov** [comment]
I dont have a mac with m1 to implement ant check the  fix

**Comment by gbrabelo** [comment]
the problem actually happens with an intel mac (it seems it is being downloaded the m1 file for a regular mac)

**Comment by SergeyPirogov** [comment]
The place to fix is here https://github.com/SergeyPirogov/webdriver_manager/blob/96446bab026f637b9e8433975631b0bec443bc99/webdriver_manager/driver.py#L100

**Comment by scousens** [comment]
in GeckoDriver.get_url(), after fetching the assets, this is what we get.
We want #5 ('name': 'geckodriver-v0.29.1-macos.tar.gz') but get #4 ('name': 'geckodriver-v0.29.1-macos-aarch64.tar.gz') as
the name.startswith matches both and we get the first found.
On Mac Mojave
{code}
result = {list: 8} [{'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696753', 'id': 34696753, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzUz', 'name': 'geckodriver-v0.29.1-linux32.tar.gz', 'label': None, 'uploader': {'login': 'whimboo', 'id': 129603,
 0 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696753', 'id': 34696753, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzUz', 'name': 'geckodriver-v0.29.1-linux32.tar.gz', 'label': None, 'uploader': {'login': 'whimboo', 'id': 129603, 
 1 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696758', 'id': 34696758, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzU4', 'name': 'geckodriver-v0.29.1-linux32.tar.gz.asc', 'label': None, 'uploader': {'login': 'whimboo', 'id': 1296
 2 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696759', 'id': 34696759, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzU5', 'name': 'geckodriver-v0.29.1-linux64.tar.gz', 'label': None, 'uploader': {'login': 'whimboo', 'id': 129603, 
 3 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696761', 'id': 34696761, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzYx', 'name': 'geckodriver-v0.29.1-linux64.tar.gz.asc', 'label': None, 'uploader': {'login': 'whimboo', 'id': 1296
 4 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696762', 'id': 34696762, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzYy', 'name': 'geckodriver-v0.29.1-macos-aarch64.tar.gz', 'label': None, 'uploader': {'login': 'whimboo', 'id': 12
 5 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696766', 'id': 34696766, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzY2', 'name': 'geckodriver-v0.29.1-macos.tar.gz', 'label': None, 'uploader': {'login': 'whimboo', 'id': 129603, 'n
 6 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696768', 'id': 34696768, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2NzY4', 'name': 'geckodriver-v0.29.1-win32.zip', 'label': None, 'uploader': {'login': 'whimboo', 'id': 129603, 'node
 7 = {dict: 13} {'url': 'https://api.github.com/repos/mozilla/geckodriver/releases/assets/34696770', 'id': 34696770, 'node_id': 'MDEyOlJlbGVhc2VBc3NldDM0Njk2Nzcw', 'name': 'geckodriver-v0.29.1-win64.zip', 'label': None, 'uploader': {'login': 'whimboo', 'id': 129603, 'node
{code}

**Comment by jhawgs** [comment]
I'm having the same problem on High Sierra. It looks like @scousens is right about the `startswith` thing. I envision a solution at the line @SergeyPirogov pointed out looking something like this:
``` python
name = f"{self.get_name()}-{self.get_version()}-{self.get_os_type()}{'-aarch64' if is_m1() else ''}" + "."
```
With this relatively simple replacement, we add the `-aarch64` specifier only if the processor is m1. Then we add the `.` to make sure that `geckodriver-v0.29.1-macos.` doesn't match `geckodriver-v0.29.1-macos-aarch64.` I'm not quite sure how this will interact with Rosetta, but it should at least make the situation a little better. I'm also not quite sure how to get whether the system is on m1, so `is_m1()` is just a placeholder. Anyone who figures that out feel free to add it in and PR.

Also, any similar PR would likely be able to close this issue as well as #174.

**Comment by SergeyPirogov** [comment]
fixed in 3.4.0
